src/langchain_google_firestore/utility/document_converter.py

```diff
@@ -0,0 +1,179 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import ast
+
+from typing import (
+    TYPE_CHECKING,
+    List,
+)
+
+from langchain_core.documents import Document
+
+IMPORT_ERROR_MSG = (
+    "`firestore` package not found, please run `pip3 install google-cloud-firestore`"
+)
+
+if TYPE_CHECKING:
+    from google.cloud.firestore_v1.client import Client
+    from google.cloud.firestore_v1.base_document import DocumentSnapshot
+    from google.cloud.firestore_v1.document import DocumentReference
+
+
+class DocumentConverter:
+
+    @staticmethod
+    def convertFirestoreDocument(
+        document: DocumentSnapshot,
+        page_content_fields: List[str] = None,
+        metadata_fields: List[str] = None,
+    ) -> Document:
+        data_doc = document.to_dict()
+        metadata = {"reference": {"path": document.reference.path}}
+
+        if page_content_fields:
+            set_page_fields = set(page_content_fields)
+        else:
+            set_page_fields = set()
+        if metadata_fields:
+            set_metadata_fields = set(metadata_fields)
+        else:
+            set_metadata_fields = set()
+        shared_keys = set_metadata_fields & set_page_fields
+
+        page_content = {}
+        for k in sorted(shared_keys):
```

Comment:
The code don't need this, but for the testing is failing because the dictionary keys are not sorted and constructing the `page_content` results in different strings. Which causes tests to fail intermittently.

e.g.
"{'f1':1, 'f2':2}"   -- sometimes is "{'f2':2, 'f1':1}"

I could remove the sort but then tests would be flaky.

Reply:
As an alternative I could complicate the tests for this comparison. Either stop using the TestCase utility that simplifies most of the tests and add for loops that extracts and converts to dict and then compare. Or need to investigate how to add some comparison function and have a variant of TestCase method used.